### PR TITLE
[web-server] Update dependency sift to v8

### LINKED ISSUE #1046: [web-server] Update dependency sift to v8
From the release notes:

> DEPRECATED sift(query, array). You must now use array.filter(sift(query)).

This will involve changing all sift invocations and making sure .filter is not called on an undefined/null array.

### PR DESCRIPTION
**Pull request by projectyang**
Fixes #1046

### CODE REVIEW

**Comment by projectyang** [comment]
Hmmm, I'm not too familiar with codecov. I ended up writing a small utility function to help with the new sift usage. Does this mean I should write a test to cover it?

File: services/web-server/src/loaders/artifacts.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const DataLoader = require('dataloader');
-const sift = require('sift').default;
+const siftUtil = require('../utils/siftUtil');
```
Comment (helfi92) [inline services/web-server/src/loaders/artifacts.js]: We can rename the utility to `sift`:

```jsx
const sift = require('../utils/sift');
```

This comment applies to all the other places we're importing this utility.

File: services/web-server/src/utils/siftUtil.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const sift = require('sift').default;
+
+// Utility function for guarding against undefined/null arrays when using sift
+module.exports = (filter, array) => {
+  if (!array) {
+    return [];
+  }
+  return filter ? array.filter(sift(filter)) : array;
+};
```
Comment (helfi92) [inline services/web-server/src/utils/siftUtil.js]: Nice!

**Comment by helfi92** [comment]
Just tested the implementation. It works :)

**Pull request merged**